From 7301914755f5843f1095ca31114149a3be7d4ba8 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Fri, 6 Oct 2023 12:31:29 +0300
Subject: [PATCH 116/834] dri: Remove __driDriverExtensions leftovers

Android-14/clang-17 throws an error with it:

ld.lld: error: version script assignment of 'global' to symbol
   '__driDriverExtensions' failed: symbol not defined

Fixes: d43e6a9a497f ("dri: Remove the megadriver compat stub")
Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
Acked-by: Erik Faye-Lund <erik.faye-lund@collabora.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25587>
---
 include/GL/internal/dri_interface.h | 8 --------
 src/gallium/targets/dri/dri.sym     | 1 -
 src/loader/loader.c                 | 2 --
 3 files changed, 11 deletions(-)

diff --git a/include/GL/internal/dri_interface.h b/include/GL/internal/dri_interface.h
index a59d598c3bf..1424e24fe27 100644
--- a/include/GL/internal/dri_interface.h
+++ b/include/GL/internal/dri_interface.h
@@ -678,14 +678,6 @@ struct __DRIuseInvalidateExtensionRec {
    __DRIextension base;
 };
 
-/**
- * The remaining extensions describe driver extensions, immediately
- * available interfaces provided by the driver.  To start using the
- * driver, dlsym() for the __DRI_DRIVER_EXTENSIONS symbol and look for
- * the extension you need in the array.
- */
-#define __DRI_DRIVER_EXTENSIONS "__driDriverExtensions"
-
 /**
  * This symbol replaces the __DRI_DRIVER_EXTENSIONS symbol, and will be
  * suffixed by "_drivername", allowing multiple drivers to be built into one
diff --git a/src/gallium/targets/dri/dri.sym b/src/gallium/targets/dri/dri.sym
index d4ffb588e05..6f502fa9f92 100644
--- a/src/gallium/targets/dri/dri.sym
+++ b/src/gallium/targets/dri/dri.sym
@@ -1,6 +1,5 @@
 {
 	global:
-		__driDriverExtensions;
 		__driDriverGetExtensions*;
 		nouveau_drm_screen_create;
 		radeon_drm_winsys_create;
diff --git a/src/loader/loader.c b/src/loader/loader.c
index a00f16f0e99..f340b61433d 100644
--- a/src/loader/loader.c
+++ b/src/loader/loader.c
@@ -836,8 +836,6 @@ loader_open_driver(const char *driver_name,
       free(get_extensions_name);
    }
 
-   if (!extensions)
-      extensions = dlsym(driver, __DRI_DRIVER_EXTENSIONS);
    if (extensions == NULL) {
       log_(_LOADER_WARNING,
            "MESA-LOADER: driver exports no extensions (%s)\n", dlerror());
-- 
2.42.0

